### PR TITLE
Stain Normalization with Macenko's method

### PR DESCRIPTION
**Pull request by nicolebussola**
## Description
Add Stain Normalization with Macenko's method

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes: #288

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/histolab/histolab/blob/master/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: histolab/stain_normalizer.py [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+        Raises
+        ------
+        ValueError
+            if ``stains`` is not a two-stains list
+        ValueError
+            if the input image is not RGB or RGBA
+        """
+
+        if stains is not None and len(stains) != 2:
```
Comment (nicolebussola) [inline histolab/stain_normalizer.py:192]: image

File: histolab/mixins.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        arr : np.ndarray
+            Input array
+
+        Returns
+        -------
+        np.ndarray
+            Two principal components.
+        """
+        _, V = np.linalg.eigh(np.cov(arr, rowvar=False))
+        return V[:, 1:3]
```
Comment (nicolebussola) [inline histolab/mixins.py:44]: image

File: histolab/mixins.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+
+class LinalgMixin:
+    @staticmethod
+    def normalize_columns(arr: np.ndarray) -> np.ndarray:
+        """Normalize each column vector in an array"""
+        return arr / np.linalg.norm(arr, axis=0)
+
+    @staticmethod
+    def two_principal_components(arr: np.ndarray) -> np.ndarray:
```
Comment (nicolebussola) [inline histolab/mixins.py:29]: image
image

Comment (nicolebussola) [reply histolab/mixins.py:29]: We will stick with the implementation using the covariance matrix (method 1) as already implemented. The svd approach (method 2) will be considered if needed

Comment (ernestoarbitrio) [inline histolab/mixins.py:29]:
```python
def principal_components(arr, start, end):
     _, V = np.linalg.eigh(np.cov(arr, rowvar=False))
     return V[:, start:end]
```

**Review by ernestoarbitrio — APPROVED** [review]

**Pull request merged**